annlite/executor.py

```diff
@@ -98,8 +110,35 @@ def index(
         if len(flat_docs) == 0:
             return
 
-        self._index.index(flat_docs)
-
+        while len(self._task_queue) >= self._max_length_queue:
+            time.sleep(1)
+
+        self._task_queue.extend(flat_docs)
+
+    def _start_index(self):
+        self._index_thread = Thread(target=self._index_task, daemon=False)
```

Comment:
please double check if is it reasonable to use`daemon=False`?

Reply:
Why is this used in a Thread?